core/api/src/main/java/org/eclipse/sensinact/core/whiteboard/WhiteboardGetSet.java

```diff
@@ -0,0 +1,68 @@
+/*********************************************************************
+* Copyright (c) 2024 Kentyou.
+*
+* This program and the accompanying materials are made
+* available under the terms of the Eclipse Public License 2.0
+* which is available at https://www.eclipse.org/legal/epl-2.0/
+*
+* SPDX-License-Identifier: EPL-2.0
+*
+* Contributors:
+*   Thomas Calmant (Kentyou) - initial implementation
+**********************************************************************/
+package org.eclipse.sensinact.core.whiteboard;
+
+import java.time.Duration;
+
+import org.eclipse.sensinact.core.twin.TimedValue;
+import org.osgi.util.promise.Promise;
+import org.osgi.util.promise.PromiseFactory;
+
+/**
+ * Whiteboard service to handle resource get/set operations
+ *
+ * @param <T> Resource type
+ */
+public interface WhiteboardGetSet<T> {
```

Comment:
I think that these should probably be separate interfaces. My reasoning is that there are a lot of read-only resources and the only way I can see to indicate this is by throwing `UnsupportedOperationException`.

It would be possible to make `WhiteboardSet` extend `WhiteboardGet` and for the default `pullValue` method of `AbstractWhiteboardSet` to just return the cached value.

Reply:
That was my initial design, but I preferred to define an Abstract with a read-only behaviour by default as it also provides the detection of the resource type by reflection.
I'm not yet sure what's the easiest API here

Reply:
I think separating them is important, otherwise the runtime has no way to know if a resource should be modelled as modifiable.

Reply:
Also, I'm not sure if the resource type / action result type should be from the handler itself or from its services properties.
Having it in the properties could allow defining handlers using lambda expressions